src/Tribe/Event_Repository.php

```diff
@@ -267,4 +312,162 @@ public function filter_by_has_rsvp( $has_rsvp = true ) {
+		$event_meta_keys = array_map( [ $wpdb, '_real_escape' ], $event_meta_keys );
+		$event_meta_keys = '"' . implode( '", "', $event_meta_keys ) . '"';
+
+		// Join to the meta that relates attendees to events.
+		$this->decorated->filter_query->join( "
+				LEFT JOIN `{$wpdb->postmeta}` AS `{$alias_event}`
+					ON `{$alias_event}`.`meta_value` = `{$wpdb->posts}`.`ID`
+			", $alias_event );
+
+		$this->decorated->where_clause( "
+				`{$alias_event}`.`meta_key` IN ( {$event_meta_keys} )
+				AND `{$alias_event}`.`post_id` IN ( {$attendee_ids} )
+			" );
+	}
+
+	/**
+	 * Filters events to include only those that do not match the provided attendee(s).
+	 *
+	 * @since TBD
+	 *
+	 * @param int|array $attendee_ids The attendee(s) to filter out.
+	 */
+	public function filter_by_attendee_not_in( $attendee_ids ) {
+		global $wpdb;
+
+		$attendee_ids = (array) $attendee_ids;
+		$attendee_ids = array_map( 'absint', $attendee_ids );
+		$attendee_ids = array_unique( $attendee_ids );
+		$attendee_ids = implode( ', ', $attendee_ids );
+
+		$alias_event = 'ticket_attendee_event';
+
+		$event_meta_keys = $this->attendee_to_event_keys();
+		$event_meta_keys = array_map( [ $wpdb, '_real_escape' ], $event_meta_keys );
+		$event_meta_keys = '"' . implode( '", "', $event_meta_keys ) . '"';
+
+		$this->decorated->where_clause( "
+				NOT EXISTS (
+					SELECT 1
+					FROM
+						`{$wpdb->postmeta}` AS `sub_{$alias_event}`
+					WHERE
+						`sub_{$alias_event}`.`meta_value` = `{$wpdb->posts}`.`ID`
+						AND `sub_{$alias_event}`.`meta_key` IN ( {$event_meta_keys} )
+						AND `sub_{$alias_event}`.`post_id` IN ( {$attendee_ids} )
+					LIMIT 1
+				)
+			" );
+	}
+
+	/**
+	 * Filters events to include only those that match the provided attendee(s).
+	 *
+	 * @since TBD
+	 *
+	 * @param int|array $user_ids The user ID(s) to filter by.
+	 */
+	public function filter_by_attendee_user( $user_ids ) {
+		global $wpdb;
+
+		$user_ids = (array) $user_ids;
+		$user_ids = array_map( 'absint', $user_ids );
+		$user_ids = array_unique( $user_ids );
+		$user_ids = implode( ', ', $user_ids );
+
+		$alias_event = 'ticket_attendee_event';
+		$alias_user  = 'ticket_attendee_user';
+
+		$event_meta_keys = $this->attendee_to_event_keys();
+		$event_meta_keys = array_map( [ $wpdb, '_real_escape' ], $event_meta_keys );
+		$event_meta_keys = '"' . implode( '", "', $event_meta_keys ) . '"';
+
+		$user_meta_key = $wpdb->_real_escape( $this->attendee_to_user_key() );
+
+		// Join to the meta that relates attendees to events.
+		$this->decorated->filter_query->join( "
+				LEFT JOIN `{$wpdb->postmeta}` AS `{$alias_event}`
+					ON `{$alias_event}`.`meta_value` = `{$wpdb->posts}`.`ID`
+			", $alias_event );
+
+		// Join to the meta that relates users to attendees.
+		$this->decorated->filter_query->join( "
+				LEFT JOIN `{$wpdb->postmeta}` AS `{$alias_user}`
+					ON `{$alias_user}`.`post_id` = `{$alias_event}`.`post_id`
+			", $alias_user );
```

Comment:
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: e75f3cf332927924329a1189873dbfa3</sub>